core/services/registrysyncer/syncer.go

```diff
@@ -0,0 +1,253 @@
+package registrysyncer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+
+	kcr "github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/capabilities_registry"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	p2ptypes "github.com/smartcontractkit/chainlink/v2/core/services/p2p/types"
+	evmrelaytypes "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+)
+
+type HashedCapabilityID [32]byte
+type DonID uint32
+
+type State struct {
+	IDsToDONs         map[DonID]kcr.CapabilitiesRegistryDONInfo
+	IDsToNodes        map[p2ptypes.PeerID]kcr.CapabilitiesRegistryNodeInfo
+	IDsToCapabilities map[HashedCapabilityID]kcr.CapabilitiesRegistryCapabilityInfo
+}
+
+type Handler interface {
```

Comment:
nit: "Handler" is a pretty overused name. I wonder if there is a good synonym (I don't have any suggestions though..)

Reply:
I could call it launcher instead which I think is what CCIP wants to call it

This would then be the CapabilityLauncher, which isn't 100% terrible